File: ruby/eleicao.rb
class Eleicao

  def initialize (eleitores, votos_brancos, votos_nulos, votos_validos)

    @eleitores = eleitores
    @votos_brancos = votos_brancos
    @votos_nulos = votos_nulos
    @votos_validos = votos_validos

  end

  def numero_total_votos
    @votos_brancos + @votos_nulos + @votos_validos
  end

  def percetagem_votos_brancos
    (@votos_brancos / @eleitores) * 100
  end

  def percetagem_votos_nulos
    (@votos_nulos / @eleitores) * 100
  end

  def percetagem_votos_validos
    (@votos_validos / @eleitores) * 100
  end

  def abstencao
    ((@eleitores - numero_total_votos()) / @eleitores) * 100
  end

end



camara = Eleicao.new(ARGV[0].to_f, ARGV[1].to_f,
                     ARGV[2].to_f, ARGV[3].to_f)


# control if there more votes than people
if camara.numero_total_votos() > ARGV[0].to_f
  abort("There is more votes than electores")
end

print("
Numero de eleitores de um municipio #{ARGV[0].to_f}
numero total de votos:  #{camara.numero_total_votos}
percetagem de votos brancos: #{camara.percetagem_votos_brancos}%
percetagem de votos nulos: #{camara.percetagem_votos_nulos}%
percetagem de votos validos: #{camara.percetagem_votos_validos}%
Abstencao: #{camara.abstencao}%

")

File: python/eleicao.py
class Eleicao ():

    def __init__(self, eleitores, brancos, nulos, validos):
        self.eleitores = eleitores
        self.brancos = brancos
        self.nulos = nulos
        self.validos = validos

    def votos_total(self):
        return self.brancos + self.nulos + self.validos

    def per_brancos(self):
        return (self.brancos / self.eleitores) * 100

    def per_nulos(self):
        return (self.nulos / self.eleitores) * 100

    def per_validos(self):
        return (self.validos / self.eleitores) * 100

    def abstencao(self):
        return ((self.eleitores - self.votos_total()) / self.eleitores) * 100


def main():
    camara = Eleicao(0, 0, 0, 0)
    camara.eleitores = float(input("Numero de eleitores: "))
    camara.brancos = float(input("Numero de votos brancos: "))
    camara.nulos = float(input("Numero de votos nulos: "))
    camara.validos = float(input("Numero de votos validos: "))

    print " Numero de eleitores de um municipio: ", camara.eleitores
    print "Numero total de votos: ", camara.votos_total()
    print "Percentagem de votos brancos: ", camara.per_brancos()
    print "Percentagem de votos nulos: ", camara.per_nulos()
    print "Percentagem de votos validos: ", camara.per_validos()
    print "Percentagem de abstencao: ", camara.abstencao()

if __name__ == "__main__":
    main()
